### PR TITLE
feat(router): implement `get_total_distance`

### PR DESCRIPTION
**Pull request by GoodluckH**

### CODE REVIEW

File: src/types/router.rs [hunk 1]
```diff
@@ -164,6 +165,34 @@ pub mod engine {
...
+        /// * `path` - The path to compute the distance of. The path is
+        ///   given as a vector of [`NodeIndex`] structs.
+        ///
+        /// # Returns
+        /// The total distance of the path.
+        ///
+        /// If the path is empty, 0.0 is returned.
+        ///
+        /// If the path is invalid, -1.0 is returned.
+        pub fn get_total_distance(&self, path: &Vec<NodeIndex>) -> f32 {
```
Comment (amsmith-pro) [inline src/types/router.rs]: Does it make more sense to return a Result<f32, Error> instead of returning -1 for invalid paths, then the caller can discard error types? A bit semantic but I get nervous seeing floating arbitrary values/magic numbers

File: src/types/router.rs [hunk 2]
```diff
@@ -529,4 +561,20 @@ mod router_tests {
         assert_eq!(edges[0].to.get_uid(), "2");
         assert_eq!(edges[1].to.get_uid(), "3");
     }
+
+    /// Test get_total_distance
+    #[test]
+    fn test_get_total_distance() {
```
Comment (amsmith-pro) [inline src/types/router.rs:595]: Also test the invalid edge cases

**Review by amsmith-pro — CHANGES_REQUESTED** [review]

File: src/types/router.rs (hunk 1 above)
Comment (GoodluckH) [reply src/types/router.rs]: [7ee52f0](https://github.com/Arrow-air/lib-router/pull/11/commits/7ee52f056943f44a59510c46820108ae8217fa8c)

File: src/types/router.rs (hunk 2 above)
Comment (GoodluckH) [reply src/types/router.rs:595]: [18f1798](https://github.com/Arrow-air/lib-router/pull/11/commits/18f1798cdd885703926e6be1d7aaa486130e8870)

**Review by amsmith-pro — APPROVED** [review]
